src/troute-network/troute/AbstractRouting.py

```diff
@@ -312,10 +312,19 @@ def topobathy_df(self):
                     for mainstem_segment in self._diffusive_network_data[tw]['mainstem_segs']:
                         if mainstem_segment not in self._topobathy_df.index:
                             # Temp.Solution: when topobaty is not available, use available topobathy of the closest upstream segment
-                            upstream_mainstem_segment = set(self._diffusive_network_data[tw]['rconn'][mainstem_segment]).intersection(
-                                                                                    self._diffusive_network_data[tw]['mainstem_segs'])
-                            upstream_mainstem_segment = list(upstream_mainstem_segment)[0]
-                            temp_df = self._topobathy_df[self._topobathy_df.index==upstream_mainstem_segment]
+                            temp_df = pd.DataFrame()
+                            position_mainstem_segment =  self._diffusive_network_data[tw]['mainstem_segs'].index(mainstem_segment)
+                            position_upstream_mainstem_segment = position_mainstem_segment
+                            while temp_df.empty:               
+                                try:
+                                    position_upstream_mainstem_segment = position_upstream_mainstem_segment - 1
```

Comment:
Why do you decrease it by 1 here? Are the IDs always consecutive?

Reply:
That does not decreasing flowpath ID number but the position of stream segment in the mainstem segment list that has aligned the list from upstream to downstream.

Reply:
True that's correct. Just a very small suggestion, its more readable to write it like `position_upstream_mainstem_segment -= 1`

Reply:
Thanks and updated.